### PR TITLE
HHH-13894 - SQL query printing on WildFly

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-13894

### CODE REVIEW

**Comment by vladmihalcea** [comment]
@sebersole If you have some time, let me know if you agree with this approach of storing the SQL string value next to the `Statement` we are registering.

File: hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java [hunk 1]
```diff
@@ -370,4 +386,21 @@ private boolean hasRegistered(final HashMap resource) {
 	private boolean hasRegistered(final ArrayList resource) {
 		return resource != null && !resource.isEmpty();
 	}
+	
+	private class StatementRegistryEntry {
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java]: Seems the inner class should be static one for seemingly it doesn't need access to enclosing object.

File: hibernate-core/src/main/java/org/hibernate/resource/jdbc/ResourceRegistry.java [hunk 2]
```diff
@@ -36,6 +36,23 @@
...
+	 * @param sql SQL {@link String} value
+	 */
+	void setStatementSql(Statement statement, String sql);
+
+	/**
+	 * Get the SQL {@link String} value associated with the {@link Statement}, if any
+	 *
+	 * @param statement JDBC Statement
+	 *
+	 * @return SQL {@link String} value
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/resource/jdbc/ResourceRegistry.java]: Do we really need the `{@link String}` above? Seems it is so basic to deserve one.

File: hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java [hunk 3]
```diff
@@ -219,17 +233,19 @@ public void register(ResultSet resultSet, Statement statement) {
...
 		if ( statement != null ) {
-			HashMap<ResultSet,Object> resultSets = xref.get( statement );
+			final StatementRegistryEntry statementRegistryEntry = xref.remove( statement );
+			HashMap<ResultSet,Object> resultSets = statementRegistryEntry != null ?
+					statementRegistryEntry.resultSets : null;
 
 			// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
 			// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
-			if ( resultSets == null ) {
+			if ( resultSets == null) {
```
Comment (NathanQingyangXu) [inline hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java]: Seems the above code change is not better than before. We do need a space char here.

File: hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java (hunk 1 above)
Comment (vladmihalcea) [reply hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java]: Good catch. It's totally better to be a nested one, especially for performance and its impact on GC.

File: hibernate-core/src/main/java/org/hibernate/resource/jdbc/ResourceRegistry.java (hunk 2 above)
Comment (vladmihalcea) [reply hibernate-core/src/main/java/org/hibernate/resource/jdbc/ResourceRegistry.java]: True. I could simply replace it with a `string` instead.

File: hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java (hunk 3 above)
Comment (vladmihalcea) [reply hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java]: That's weird. I thought I ran a CTRL ALT L on those bocks, and the `check` command didn't complain either.

I'll fix all issues and push a new commit.

**Comment by vladmihalcea** [comment]
@NathanQingyangXu I pushed a new commit to address the issues you found. Thanks for taking the time to review it.

**Comment by NathanQingyangXu** [comment]
> @NathanQingyangXu I pushed a new commit to address the issues you found. Thanks for taking the time to review it.

Thanks. I have no further concern, but I guess Steve is the final reviewer so it won't matter whether I approve or not.

**Comment by NathanQingyangXu** [comment]
btw, there is another related PR recently: https://github.com/hibernate/hibernate-orm/pull/3287. Seems the 'slowQuery' feature is really popular.

**Comment by nicolas-tg-ch** [comment]
Any plans to merge this soon in Hibernate 5.4/5.5? I have the same problem on Websphere Liberty, see https://stackoverflow.com/questions/68815550/hibernates-slow-sql-queries-in-websphere-liberty

(I had many problems to login/sign up on hibernate.atlassian.net that's why I post here.)

**Comment by gavinking** [comment]
This problem was already fixed in a different PR by @NathanQingyangXu.

**Pull request closed without merging**